docs/debugging-transactions.md

```diff
@@ -0,0 +1,510 @@
+# Debugging Transactions
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Debugging with Remix](#1-debugging-with-remix)
+    - [Steps to Use the Built-in Debugger](#steps-to-use-the-built-in-debugger)
+    - [Example of Debugging a Transaction](#example-of-debugging-a-transaction)
+3. [Debugging with Truffle](#2-debugging-with-truffle)
+    - [Steps to Use the Debugger in Truffle](#steps-to-use-the-debugger-in-truffle)
+4. [Advanced Debugging with `debug_traceTransaction`](#2-advanced-debugging-with-debug_traceTransaction)
+    - [Manually Calling `debug_traceTransaction`](#manually-calling-debug_traceTransaction)
+    - [Examples of Using `debug_traceTransaction`](#examples-of-using-debug_tracetransaction)
+        - [With `callTracer`](#1-with-callTracer)
+        - [With `opcodeLogger`](#2-with-opcodeLogger)
+
+## Overview
+
+[Remix IDE](https://remix-project.org/) provides powerful debugging tools that allow developers to inspect and analyze
+the execution of smart contracts.
+
+This guide will walk you through the various debugging capabilities available, including:
+
+1. how to use the built-in debugger in Remix to step through the execution of a transaction and inspect the state of
+   your smart contracts, and
+2. how to manually call the `debug_traceTransaction` method and analyze the response to get detailed information about
+   the execution of a transaction.
+
+## 1. Debugging with Remix
+
+The built-in debugger in Remix allows you to step through the execution of a transaction, inspect the state, and
+understand the behavior of your smart contracts.
+
+### Steps to Use the Built-in Debugger
+
+Official Documentation: [Remix Debugger](https://remix-ide.readthedocs.io/en/latest/debugger.html)
+
+1. **Set Up Remix with a Local Node**:
+    - Ensure you have a local node running that supports the `debug_traceTransaction` method.
```

Comment:
Is there anything you can add here?  Like a minimum release version?  Or if there is a configuration setting?   (For the local node.)

Reply:
Hmmm, good point. We need to merge the PRs which refactor the validations and the `debug_traceTransaction` implementation, cause currently it doesn’t work without those fixes and then we will need to refer to the local node release that includes those changes.

Reply:
If you don't want to wait for that before merging this PR you could leave placeholder text here and just remember to do it (or file an issue to do it).